src/main/java/wolox/training/models/Book.java

```diff
@@ -78,14 +79,16 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
-        genre = genre;
+        Preconditions.checkNotNull(genre);
```

Comment:
After checking if a field is null, check if the field is empty.

An error message can be added to each check :

```
Preconditions.checkNotNull(genre, "The genre field cannot be null.");
Preconditions.checkArgument(!genre.isEmpty(), "The genre field cannot be empty.");
this.genre = genre;
```

I recommend using a class to store error messages as constants.

Reply:
I put the messages and added it like constants.